### PR TITLE
Add UI test and benchmark report

### PR DESCRIPTION
**Pull request by trungleduc**
## References
https://github.com/voila-dashboards/voila/issues/948

## Code changes

Add more UI tests and use playwright to benchmark performance in CI.
The CI script for benchmark reporting is taken from jupyterlab project

### CODE REVIEW

**Review by jtpio — APPROVED** [review]
Thanks!

File: ui-tests/tests/utils.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        time: average(testTimeArray)
+      })
+    );
+    await new Promise(r => setTimeout(r, 500));
+  }
+  // testInfo.attachments.push({
+  //   name: notebookName,
+  //   contentType: 'application/json',
+  //   body: Buffer.from(JSON.stringify({ testTime: average(testTimeArray) }))
+  // });
```
Comment (jtpio) [inline ui-tests/tests/utils.ts:49]: Do we need to keep this comment, or could it be removed?

Comment (trungleduc) [reply ui-tests/tests/utils.ts]: Thanks, removed

**Comment by jtpio** [comment]
The `macos` test failure is not related (tracked in #961).

@trungleduc is this ready to go or did you want to push more changes? we can iterate in separate PRs.

**Comment by trungleduc** [comment]
Yes I think it's ok to go, the report on PR comment feature will come in next PR

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**